lib/Makefile

```diff
@@ -330,7 +330,8 @@ include $(wildcard $(DEPFILES))
 # Special case : building library in single-thread mode _and_ without zstdmt_compress.c
 ZSTDMT_FILES = compress/zstdmt_compress.c
 ZSTD_NOMT_FILES = $(filter-out $(ZSTDMT_FILES),$(ZSTD_FILES))
-libzstd-nomt: LDFLAGS += -shared -fPIC -fvisibility=hidden
+libzstd-nomt: CFLAGS += -fvisibility=hidden
```

Comment:
👍

Reply:
Thanks for taking a look ;-). I’m wondering what to do about `ZSTD_getParams` since `pztd` uses that dynamically even though it’s supposed to be a static-only symbol. Declaring it as a non-static API would fix the failing tests, but that might not be a stability promise the project is happy with!

(For context, I’m trying to reduce the amount of exported symbols to avoid [issues such as this one](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=969597) — Debian carries a symbols file, which is used to generate versioned dependencies, so that a binary can have dependencies which are as “generous” as possible; but this gets confused by `libzstd`’s symbols. Perhaps the correct solution is to drop that file entirely, and always request a version of `libzstd` equal than or greater to the version with which a given binary was compiled...)

Reply:
> Perhaps the correct solution is to (...) always request a version of `libzstd` equal than or greater to the version with which a given binary was compiled

`libzstd` library policy is to only update the "stable" section of the API when the second digit changes.
For example, in `1.4.x` series, all versions from `v1.4.0` to `v1.4.8` support the exact same set of symbols as "stable".
(Meanwhile, the "experimental" section can, and does, evolve at each iteration.)

The next update to "stable" API will happen at `v1.5.0`.
Even then, we are restrained to only _add_ new symbols, existing symbols are considered completely stable, and guaranteed to remain present in all `v1.x.y` versions of the library.

Given that, the proposed rule (always request a version of `libzstd` equal than or greater to the version with which a given binary was compiled) would work.
A slightly more generous rule would allow usage of older `libzstd` as long as the second digit matches or is greater (i.e. an application compiled with `v1.4.5` would still accept a `v1.4.0` library, but not `v1.3.6`). It's effectively the same rule, but limited to the first 2 digits.

Reply:
Right, that’s what I suspected. The problem in Debian is that the default analysis results in, for example, the library package declaring that `ZSTD_minCLevel` is available since 1.3.8 (which is when it appeared in the library’s symbols) even though it’s only part of the stable API since 1.4. But that really means the problem lies in the tools used ;-).